doc/uuid.html

```diff
@@ -591,34 +597,106 @@
     explicit basic_random_generator(UniformRandomNumberGenerator&amp; gen);
     explicit basic_random_generator(UniformRandomNumberGenerator* pGen);
 
-    uuid operator()();
+    result_type operator()();
 };
-typedef basic_random_generator&lt;mt19937&gt; random_generator;
+
+    typedef basic_random_generator&lt;boost::mt19937&gt; random_generator_mt19937;
 
 }} // namespace boost::uuids
 </pre>
 
 <h4><a name="Random Generator">Random Generator</a></h4>
-<p>
-The <tt>boost::uuids::basic_random_generator</tt> class generates a random number
-based uuid from a random number generator (one that conforms to the
-<a href="http://www.boost.org/libs/random/random-concepts.html#uniform-rng">UniformRandomNumberGenerator</a>
-concept).
-<pre>
-//default construct the random number generator and seed it
-boost::uuids::basic_random_generator&lt;boost::mt19937&gt; gen;
-boost::uuids::uuid u = gen();
+    <p>
+        <tt>boost::uuids::random_generator</tt> class generates uuids using
+        operating system provided entropy.  For the majority of use cases this
+        should be sufficient, as this generator has very low startup overhead
+        when compared to a generator with seeding requirements.
+    </p>
+    <p>
+        <tt>boost::uuids::basic_random_generator</tt> class generates a random number
+        based uuid from a random number generator (one that conforms to the
+        <a href="http://www.boost.org/libs/random/random-concepts.html#uniform-rng">UniformRandomNumberGenerator</a>
+        concept).  The default constructor will properly seed the random number generator
+        if it requires the behavior for proper operation.  Additional constructors allow you
+        to provide your own <tt>UniformRandomNumberGenerator</tt> and you are responsible for
+        properly seeding it if necessary.
+    </p>
+    <p>
+        <tt>boost::uuids::random_generator_mt19937</tt> is a type definition for:
+        <tt>boost::uuids::basic_random_generator&lt;mt19937&gt;</tt> and is provided for
+        convenience.
+    </p>
+
+<h5>Platforms</h5>
+    <p>
+        The following platforms are supported and entropy selection logic explained:
+        <ul>
+            <li>CloudABI</li>
+            <li>Unix: use <tt>getentropy(3)</tt> if detected, otherwise use <tt>/dev/urandom</tt>, otherwise use <tt>/dev/random</tt>
+            <li>Windows (UWP Compatible): use BCrypt if available, otherwise use Wincrypt
+</ul>
+    </p>
 
-//for convenience boost::uuids::random_generator
-//is equivalent to boost::uuids::basic_random_generator&lt;boost::mt19937&gt;
+<h5>Preprocessor Macros</h5>
+    <p>
+        <ul>
+            <li>
+                <tt>BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX</tt>: on Unix this will force the
+                selection of <tt>/dev/*random</tt> over getentropy(3).
```

Comment:
Prefer to not provide this. By all means give people a macro to override behavior *when they ask for it*, no need to to it preemptively.

Reply:
I'm leveraging this to get coverage and a (mostly) common sad path test.  I'll look at testing against the posix.ipp implementation directly when running on linux.

Reply:
Yes, I hadn't yet gotten to the testing part. Keep those macros. I was looking at them from the user side.

Reply:
That'll save a bunch of additional work... thanks.